drop-boxes/register-nanopore-dropbox/register-nanopore.py

```diff
@@ -0,0 +1,281 @@
+SERVICE_CREDENTIALS.password = tracking_helper.get_service_password()
+SERVICE_REGISTRY_URL = URL(tracking_helper.get_service_reg_url())
+QBIC_LOCATION = tracking_helper.get_qbic_location_json()
+
+### We need this object to update the sample location later
+SAMPLE_TRACKER = SampleTracker.createQBiCSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, QBIC_LOCATION)
+
+# ETL script for registration of VCF files
+# expected:
+# *Q[Project Code]^4[Sample No.]^3[Sample Type][Checksum]*.*
+pattern = re.compile('Q\w{4}[0-9]{3}[a-xA-X]\w')
+alt_pattern = re.compile('Q\w{4}ENTITY-[1-9][0-9]*')
+
+NANOPORE_EXP_TYPE_CODE = "Q_NGS_NANOPORE_RUN"
+NANOPORE_SAMPLE_TYPE_CODE = "Q_NGS_NANOPORE_SINGLE_SAMPLE_RUN"
+NANOPORE_LOG_CODE = "Q_NGS_NANOPORE_RUN_LOGS"
+NANOPORE_FASTQ_CODE = "Q_NGS_NANOPORE_RUN_FASTQ"
+NANOPORE_FAST5_CODE = "Q_NGS_NANOPORE_RUN_FAST5"
+NANOPORE_SAMPLE_PREFIX = "NGS"
+
+# needed for pooled samples with multiple measurements
+usedSampleIdentifiers = set()
+usedExperimentIdentifiers = set()
+
+def createNewSample(transaction, space, parentSampleCode):
+    run = 0
+    sampleExists = True
+    newSampleID = None
+    while sampleExists:
+        run += 1
+        newSampleID = '/' + space + '/' + NANOPORE_SAMPLE_PREFIX + str(run) + parentSampleCode
+        sampleExists = transaction.getSampleForUpdate(newSampleID) or newSampleID in usedSampleIdentifiers
+    usedSampleIdentifiers.append(newSampleID)
+    return transaction.createNewSample(newSampleID, NANOPORE_SAMPLE_TYPE_CODE)
+
+def createNewExperiment(transaction, space, project):
+    search_service = transaction.getSearchService()
+    existingExperiments = search_service.listExperiments("/" + space + "/" + project)
+    for eexp in existingExperiments:
+        usedExperimentIdentifiers.append(eexp.getExperimentIdentifier())
+    run = len(usedExperimentIdentifiers)
+    expExists = True
+    newExpID = None
+    while expExists:
+        run += 1
+        newExpID = '/' + space + '/' +project+ '/' + project+str(run)
+        expExists = newExpID in usedExperimentIdentifiers
+    usedExperimentIdentifiers.append(newExpID)
+    return transaction.createNewExperiment(newExpID, NANOPORE_EXP_TYPE_CODE)
+
+# return first line of file
+def getDatahandlerMetadata(incomingPath, fileName):
+    path = os.path.realpath(os.path.join(incomingPath,file_name))
+    with open(path) as f:
+        return f.readline()
+
+def getTimeStamp():
+    now = datetime.now()
+    ts = str(now.minute)+str(now.second)+str(now.microsecond)
+    return ts
+
+# return absolute path of copied folder containing all log files. used to add log files to each involved sample
+def copyLogs(parentPath, fileList):
+    ts = getTimeStamp()
+    newLogFolder = os.path.join(parentPath, ts+"/logs")
+    os.mkdir(newLogFolder)
+    for logFile in fileList:
+        src = os.path.join(parentPath, logFile)
+        shutil.copy2(src, newLogFolder)
+    return newLogFolder
```

Comment:
I suggest to make the function more focused, and have a separate method for the log dir creation:

```suggestion
def copyLogFilesTo(logFiles, filePath, targetFolderPath):
    for logFile in logFiles:
        src = os.path.join(filePath, logFile)
        shutil.copy2(src, targetFolderPath)
    copiedContent = os.listdir(targetFolderPath)
    if len(copiedContent) != len(logFiles):
        raise AssertionError("Not all log files have been copied successfully to target log folder.")

def createLogFolder(targetPath):
    ts = getTimeStamp()
    newLogFolder = os.path.join(targetPath, os.path.join(ts, "logs"))
    os.mkdir(newLogFolder)
    return newLogFolder
```

Reply:
changed in latest commits